pylibczi/tests/test_czi_file.py

```diff
@@ -0,0 +1,100 @@
+import pytest
+from pylibczi import CziFile
+from pathlib import Path
+
+
+@pytest.mark.parametrize("fname", [
+    pytest.param('', marks=pytest.mark.raises(exception=IsADirectoryError))
+])
+def test_is_a_directory(data_dir, fname):
+    czi = CziFile(str(data_dir))
+
+
+@pytest.mark.parametrize("fname, xp_query, expected", [
+    pytest.param('s_1_t_1_c_1_z_1.czi', ".//SizeS", 1, marks=pytest.mark.raises(exception=AttributeError)),
+    pytest.param('s_1_t_1_c_1_z_1.czi', ".//SizeZ", 1, marks=pytest.mark.raises(exception=AttributeError)),
+    pytest.param('s_1_t_1_c_1_z_1.czi', ".//SizeT", 1, marks=pytest.mark.raises(exception=AttributeError)),
+    ('s_1_t_1_c_1_z_1.czi', ".//SizeC", 1),
+    ('s_1_t_1_c_1_z_1.czi', ".//SizeY", 325),
+    ('s_1_t_1_c_1_z_1.czi', ".//SizeX", 475),
+    ('s_3_t_1_c_3_z_5.czi', ".//SizeS", 3),
+    pytest.param('s_3_t_1_c_3_z_5.czi', ".//SizeT", 1, marks=pytest.mark.raises(exception=AttributeError)),
+    ('s_3_t_1_c_3_z_5.czi', ".//SizeC", 3),
+    ('s_3_t_1_c_3_z_5.czi', ".//SizeZ", 5),
+])
+def test_metadata(data_dir, fname, xp_query, expected):
```

Comment:
Love it. Absolutely love it. Once this is implemented and hooked up to `aicsimageio` we can finally close https://github.com/AllenCellModeling/aicsimageio/issues/4